cmd/minikube/cmd/start.go

```diff
@@ -1419,6 +1423,10 @@ func validateKubernetesVersion(old *config.ClusterConfig) {
 		exitIfNotForced(reason.KubernetesTooOld, "Kubernetes {{.version}} is not supported by this release of minikube", out.V{"version": nvs})
 	}
 
+	if nvs.GT(newestVersion) {
```

Comment:
thank you @JacekDuszenko for this PR ! while we don't support k8s versions newer than newest we still allow users to set a a Newer kubernetes version.
in this case I suggest in if the k8s version is Newer than our Support Newest version, we check the Kubernetes releases and see if that is an actual released version if not exit
if yes allow the user to do it at their own risk

Reply:
@medyagh Can you please elaborate on how I should check the k8s releases? Is there a bit of code somewhere in minikube that already does that? If not, what's the best place to find releases? Should I make a REST call to some kind of a web service or rather try to parse kubernetes docs page where all releases all listed?

Reply:
Parsing the available versions sounds like a totally different feature. I just think you need to make version higher (like 1.26)

Reply:
@medyagh I found the code that fetches versions myself and implemented fetching and checking the version according to your proposal. This is updated and ready to be reviewed now

Reply:
@afbjorklund thanks! So now we could either:
1. Withdraw parsing available versions feature and make version higher.
2. Leave parsing versions as they are now and make version higher.
3. Do not make the version higher as I think with fetching available releases and parsing them the issue is resolved? 

What should we do then?

Reply:
Parsing the available version is a good feature, just thought it was a _different_ one

Reply:
This needs to have some kind of offline override, when running in an air-gapped installation.

Reply:
@JacekDuszenko here is one place
https://github.com/medyagh/minikube/blob/5331cbeb8f1664ba69335e08e688d635c1767e37/hack/update/kubernetes_version/update_kubernetes_version.go#L148

Reply:
Thanks @medyagh. I've already found code that does that. It would be really useful for me to have you look at this chain of comments: https://github.com/kubernetes/minikube/pull/11631#discussion_r650674494 and let us know if we definitely want to go with the feature flag approach here. Thanks once again!